### ISSUE TITLE
EVReflection, Realm, and Primitive Arrays

### ISSUE THREAD

**Issue by Alarson93** [evermeer/EVReflection#244]
Let's say I have some JSON. A key, "urls" contains a list of strings (that are links).

I am also using Realm in this example and would like to save this object to my database. Realm does not have support for array's of primitives; thus, I must wrap my values into something like:

`class RealmString: Object {
    @objc dynamic var value = ""
}`

It is also worth noting that there are many other keys/properties on this object that work fine with reflection; "urls" is the odd one. I assume PropertyConvertors are the right way to go, but I can't seem to override that function on a Realm Object.
Any help you could offer to point me in the right direction would be appreciated!

**Comment by evermeer** [comment]
On Realm Object is extended with the EVReflectable protocol with default implementations. This means that if you want to implement a function you don't have to overwrite, you just have to define the function. In your case the function will look something like (untested):

```
    open func propertyConverters() -> [(key: String, decodeConverter: ((Any?) -> ()), encodeConverter: (() -> Any?))] {
        return [(key: "url", decodeConverter: { stringArray in
            self.urls = (stringArray as! NSArray).map { StringObject(value: $0) }
        }, encodeConverter: { 
            return self.urls.map { $0.value }
        })]
    }
```

**Comment by Alarson93** [comment]
I've added the code you've provided onto my model object. Some tweaking might be needed of course, but I figured I might as well see how it works as-is. Unfortunately, the propertyConverters() function is never called.

**Comment by evermeer** [comment]
@Alarson93 .. Ay.. just found out that my last change indeed broke handling implementing these kind of functions. In the previous version I moved the EVReflectable protocol as a default on Object. But doing that you won't be able to override the default protocol implementation functions. So it's now removed from that. The downside is that you have to add EVReflectable to each Object class definition. See the Realm unit tests in the project. It's now released as version 5.1.0

**Comment by Alarson93** [comment]
Thank you very much! I updated my pods this morning and your fix has resolved my issue :)
